sys/targets/targets.go

```diff
@@ -38,6 +38,7 @@ type Target struct {
 	NeedSyscallDefine  func(nr uint64) bool
 	HostEndian         binary.ByteOrder
 	SyscallTrampolines map[string]string
+	Target32           string // the 32-bit pair of this target
```

Comment:
This looks too global for the goal and the mapping is duplicated across all OSes.
I would just map it locally where  you need it as:
```
map[string]string{
	AMD64: I386,
	ARM64: ARM,
}[goarch]
```